### PR TITLE
Heroku

### PR DESCRIPTION
**Pull request by jonnyarnold**
Another pull request test. Come on, you'll work now Samuel.

### CODE REVIEW

**Comment by lpil** [comment]
Squish plz.

File: lib/samuel/data_provider.ex [hunk 1]
```diff
@@ -58,14 +62,33 @@ defmodule Samuel.DataProvider do
...
     headers = Map.merge(default_headers, headers)
     response = http.get!(url, headers)
-    Poison.Parser.parse!(response.body)
+    parse(response.body, response.headers)
+  end
+
+  defp parse(body, headers) when is_list(headers) do
+    content_type = headers |> Enum.find_value(fn
+      {"Content-Type", type} -> type
+      _ -> nil
```
Comment (lpil) [inline lib/samuel/data_provider.ex]: I would like to find a way to clean this up a bit.

File: lib/samuel/dispatchers/github.ex [hunk 2]
```diff
@@ -15,8 +15,10 @@ defmodule Samuel.Dispatchers.Github do
     pull = action.pull_id
     json = Poison.encode!(%{ body: action.message })
 
-    "https://api.github.com/repos/#{repo}/issues/#{pull}/comments"
-    |> http.post!(auth_headers, json)
+    response = "https://api.github.com/repos/#{repo}/issues/#{pull}/comments"
+    |> http.post!(json, auth_headers)
+
+    response
```
Comment (lpil) [inline lib/samuel/dispatchers/github.ex]: Why?

File: test/samuel/data_provider_test.exs [hunk 3]
```diff
@@ -60,17 +73,27 @@ defmodule Samuel.DataProviderTest do
             "comments_url" => "COMMENTS-URL"
           }
         }
-        data = DataProvider.fetch_requirements(requirements, event, HTTPClient)
-        assert data.comments == "DATA-FOR-COMMENTS-URL"
+        data = DataProvider.fetch_requirements(requirements, event, JsonClient)
```
Comment (lpil) [inline test/samuel/data_provider_test.exs]: JSONClient to match HTTPClient

File: test/samuel/data_provider_test.exs [hunk 4]
```diff
@@ -60,17 +73,27 @@ defmodule Samuel.DataProviderTest do
...
+          }
+        end
+      end
+
       should "look up the comments URL from the event and fetch data" do
         requirements = [:guidelines]
         event = nil
-        data = DataProvider.fetch_requirements(requirements, event, HTTPClient)
-        assert data.guidelines == "DATA-FOR-https://api.github.com/repos/"
+        data = DataProvider.fetch_requirements(requirements, event, MarkdownClient)
```
Comment (lpil) [inline test/samuel/data_provider_test.exs]: TextClient? Could reuse it

File: lib/samuel/data_provider.ex (hunk 1 above)
Comment (jonnyarnold) [inline lib/samuel/data_provider.ex]: Me too :)

File: test/samuel/data_provider_test.exs (hunk 4 above)
Comment (lpil) [inline test/samuel/data_provider_test.exs]: Oops, it sets the markdown content type. Nevermind.

**Pull request merged**